data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/HistogramAggregateActionTests.java

```diff
@@ -235,4 +240,132 @@ void testHistogramAggregateOTelFormat(int testCount) throws NoSuchFieldException
             }
         }
     }
+
+    @ParameterizedTest
+    @ValueSource(ints = {10, 20, 50, 100})
+    void testHistogramAggregateOTelFormatWithStartAndEndTimesInTheEvent(int testCount) throws NoSuchFieldException, IllegalAccessException {
+        HistogramAggregateActionConfig mockConfig = mock(HistogramAggregateActionConfig.class);
```

Comment:
This test is hard to read. Might be worth creating some private functions to make it more readable.